### PR TITLE
docs(nodejs): fix `devDependencies` installation explanation

### LINKED ISSUE #2645: [Javascript] devDependencies documentation is incorrect, they are now installed by default
Our nodejs-buildpack and our NodeJS documentation https://doc.scalingo.com/languages/nodejs/start#install-devdependencies

It seems our documentation does not match reality on the handling of devDependencies.
The buildpack behaviour has changed, devDependencies are always installed, then they are pruned at the end of the build (so only production dependencies are left during the runtime)

Our documentation states the opposite and should be updated.

### PR DESCRIPTION
**Pull request by EtienneM**
Fix #2645

### CODE REVIEW

File: src/_posts/languages/nodejs/2000-01-01-start.md [hunk 1]
```diff
@@ -92,19 +92,11 @@ Buildpack]({% post_url platform/deployment/buildpacks/2000-01-01-ssh-key %}),
...
-variable to false via our web interface or by using our CLI:
+By default, dependencies present in the `devDependencies` field are installed. At the end of the deployment, `devDependencies` are pruned. Hence only production dependencies are left in the image used for the runtime.
 
-```bash
-$ scalingo --app my-app env-set NPM_CONFIG_PRODUCTION=false
-```
-
-After setting the NPM_CONFIG_PRODUCTION environment variable, you have to
-redeploy your code.
+The pruning step can be skipped by setting the environment variable `YARN_PRODUCTION` to any value.
```
Comment (mcimbaro) [inline src/_posts/languages/nodejs/2000-01-01-start.md:99]: Sure? Setting `YARN_PRODUCTION` to `true` shouldn't skip

Comment (EtienneM) [reply src/_posts/languages/nodejs/2000-01-01-start.md:99]: I'm pretty sure, yes:

https://github.com/Scalingo/nodejs-buildpack/blob/5169a57250df80a532f1aa9c4e19b8bb18ed3b46/lib/dependencies.sh#L176-L179

Comment (mcimbaro) [reply src/_posts/languages/nodejs/2000-01-01-start.md:99]: Seems a bit counter-intuitive, but you're right

**Review by mcimbaro — APPROVED** [review]

File: src/_posts/languages/nodejs/2000-01-01-start.md (hunk 1 above)
Comment (EtienneM) [reply src/_posts/languages/nodejs/2000-01-01-start.md:99]: Thanks for paying attention to this detail :)

**Pull request merged**